rbd/pool_nautilus.go

```diff
@@ -84,3 +84,151 @@ func RemovePoolMetadata(ioctx *rados.IOContext, key string) error {
+// Call PoolStats.Destroy to free the resources.
+//
+// Implements:
+//  void rbd_pool_stats_create(rbd_pool_stats_t *stats)
+func PoolStatsCreate() *PoolStats {
+	poolStats := &PoolStats{}
+	C.rbd_pool_stats_create(&poolStats.stats)
+	return poolStats
+}
+
+// Destroy a PoolStats struct and free the associated resources.
+//
+// Implements:
+//  void rbd_pool_stats_destroy(rbd_pool_stats_t stats)
+func (poolStats *PoolStats) Destroy() {
+	C.rbd_pool_stats_destroy(poolStats.stats)
+
+	if poolStats.stats != nil {
+		poolStats.stats = nil
+	}
+}
+
+// PoolStatOption represents a group of configurable pool stat options.
+type PoolStatOption C.rbd_pool_stat_option_t
+
+const (
+	// PoolStatOptionImages is the representation of
+	// RBD_POOL_STAT_OPTION_IMAGES from librbd.
+	PoolStatOptionImages = PoolStatOption(C.RBD_POOL_STAT_OPTION_IMAGES)
+	// PoolStatOptionImageProvisionedBytes is the representation of
+	// RBD_POOL_STAT_OPTION_IMAGE_PROVISIONED_BYTES from librbd.
+	PoolStatOptionImageProvisionedBytes = PoolStatOption(C.RBD_POOL_STAT_OPTION_IMAGE_PROVISIONED_BYTES)
+	// PoolStatOptionImageMaxProvisionedBytes is the representation of
+	// RBD_POOL_STAT_OPTION_IMAGE_MAX_PROVISIONED_BYTES from librbd.
+	PoolStatOptionImageMaxProvisionedBytes = PoolStatOption(C.RBD_POOL_STAT_OPTION_IMAGE_MAX_PROVISIONED_BYTES)
+	// PoolStatOptionImageSnapshots is the representation of
+	// RBD_POOL_STAT_OPTION_IMAGE_SNAPSHOTS from librbd.
+	PoolStatOptionImageSnapshots = PoolStatOption(C.RBD_POOL_STAT_OPTION_IMAGE_SNAPSHOTS)
+	// PoolStatOptionTrashImages is the representation of
+	// RBD_POOL_STAT_OPTION_TRASH_IMAGES from librbd.
+	PoolStatOptionTrashImages = PoolStatOption(C.RBD_POOL_STAT_OPTION_TRASH_IMAGES)
+	// PoolStatOptionTrashProvisionedBytes is the representation of
+	// RBD_POOL_STAT_OPTION_TRASH_PROVISIONED_BYTES from librbd.
+	PoolStatOptionTrashProvisionedBytes = PoolStatOption(C.RBD_POOL_STAT_OPTION_TRASH_PROVISIONED_BYTES)
+	// PoolStatOptionTrashMaxProvisionedBytes is the representation of
+	// RBD_POOL_STAT_OPTION_TRASH_MAX_PROVISIONED_BYTES from librbd.
+	PoolStatOptionTrashMaxProvisionedBytes = PoolStatOption(C.RBD_POOL_STAT_OPTION_TRASH_MAX_PROVISIONED_BYTES)
+	// PoolStatOptionTrashSnapshots is the representation of
+	// RBD_POOL_STAT_OPTION_TRASH_SNAPSHOTS from librbd.
+	PoolStatOptionTrashSnapshots = PoolStatOption(C.RBD_POOL_STAT_OPTION_TRASH_SNAPSHOTS)
+)
+
+// addPoolStatOption adds the given PoolStatOption to PoolStats.
+//
+// Implements:
+//  int rbd_pool_stats_option_add_uint64(rbd_pool_stats_t stats, int stat_option, uint64_t* stat_val)
+func (poolStats *PoolStats) addPoolStatOption(option PoolStatOption, val *uint64) error {
+	ret := C.rbd_pool_stats_option_add_uint64(
+		poolStats.stats,
+		C.int(option),
+		(*C.uint64_t)(val))
+	return getError(ret)
+}
+
+// GetPoolStats returns the value for the PoolStatOption.
+func (poolStats *PoolStats) GetPoolStats(ioctx *rados.IOContext, option PoolStatOption) (uint64, error) {
```

Comment:
Interesting, but like the type being exported this seem like it might not provide a lot of value to the caller, versus just using GetAllPoolStats and just grabbing the one value you care about out of the map. I'd argue that we should skip this for now but if there's demand for it at a later time (say some sort of measurable performance difference) we can add it later.

Reply:
Agree, will remove it. Wanted to get an opinion based on the last conversation.